src/backend/mod.rs

```diff
@@ -1,30 +1,57 @@
-//! This module provides the backend implementations for different terminal libraries.
-//! It defines the [`Backend`] trait which is used to abstract over the specific
-//! terminal library being used.
+//! Provides [`Backend`] implementations for different terminal libraries.
 //!
-//! The following terminal libraries are supported:
-//! - Crossterm (with the `crossterm` feature)
-//! - Termion (with the `termion` feature)
-//! - Termwiz (with the `termwiz` feature)
+//! This module defines the [`Backend`] trait which is used to abstract over the specific terminal
+//! library being used.
+//!
+//! Supported terminal backends:
+//! - [Crossterm]: enable the `crossterm` feature (enabled by default) and use [`CrosstermBackend`]
+//! - [Termion]: enable the `termion` feature and use [`TermionBackend`]
+//! - [Termwiz]: enable the `termwiz` feature and use [`TermwizBackend`]
 //!
 //! Additionally, a [`TestBackend`] is provided for testing purposes.
 //!
 //! # Example
 //!
-//! ```rust
+//! ```rust,no_run
 //! use ratatui::backend::{Backend, CrosstermBackend};
 //!
-//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
-//! let buffer = std::io::stdout();
-//! let mut backend = CrosstermBackend::new(buffer);
+//! let mut backend = CrosstermBackend::on_stdout()?;
 //! backend.clear()?;
-//! # Ok(())
-//! # }
+//! # std::io::Result::Ok(())
 //! ```
 //!
-//! [`Backend`]: trait.Backend.html
-//! [`TestBackend`]: struct.TestBackend.html
-
+//! # Raw Mode
+//!
+//! Raw mode is a mode where the terminal does not perform any processing on the input and output.
+//! This is useful for applications that want to process the input and output themselves. For
+//! example, in raw mode, the terminal will not perform line buffering on the input, so the
+//! application will receive each key press as it is typed instead of waiting for the user to press
+//! enter.
+//!
+//! Each backend handles raw mode differently, so the behavior may vary depending on the backend
+//! being used.
+//!
+//! # Alternate Screen
+//!
+//! The alternate screen is a screen that is hidden when not in use. When an application exits, the
+//! terminal will switch back to the main screen and the contents of the alternate screen will be
+//! cleared. This is useful for applications that want to display their own output without having to
+//! manage the contents of the terminal.
+//!
+//! Each backend handles the alternate screen differently, so the behavior may vary depending on the
+//! backend being used.
+//!
+//! # Mouse Capture
+//!
+//! Mouse capture is a mode where the terminal captures mouse events and sends them to the
+//! application. This is useful for applications that want to process mouse events themselves.
+//!
+//! Each backend handles mouse capture differently, so the behavior may vary depending on the
+//! backend being used.
```

Comment:
```suggestion
//! # Mouse Capture
//!
//! Mouse capture is a mode where the terminal captures mouse events such as clicks, scrolls, and
//! movement, and sends them to the application as special sequences or events. This enables the
//! application to handle and respond to mouse actions, providing a more interactive and graphical
//! user experience within the terminal. It's particularly useful for applications like terminal-based
//! games, text editors, or other programs that require more direct interaction from the user.
//!
//! Each backend handles mouse capture differently, with variations in the types of events that can be
//! captured and how they are represented. As such, the behavior may vary depending on the backend
//! being used, and developers should consult the specific backend's documentation to understand how it
//! implements mouse capture.
```

Reply:
This is good. We probably need to add something about how handling input is not really part of Ratatui's scope, and so while we provide the means to enable this, we don't provide any code that does anything useful for the mouse capture mode.